File: Gerador de Senha/senha.cpp
#include <iostream>
#include <string>
#include <vector>
#include <iomanip>
#include <cmath>

using namespace std;

vector<string> gerarSenhasForcaBruta(int numDigitos) {
    vector<string> senhas;

    if (numDigitos <= 0) {
        cerr << "Erro: A quantidade de dígitos deve ser um número positivo." << endl;
        return senhas;
    }

    long long limiteSuperior = static_cast<long long>(pow(10, numDigitos));

    for (long long i = 0; i < limiteSuperior; ++i) {
        string senha = to_string(i);
        while (senha.length() < numDigitos) {
            senha = "0" + senha;
        }
        senhas.push_back(senha);
    }

    return senhas;
}

int main() {
    int quantidadeDigitos;

    cout << "Digite a quantidade de digitos desejada para as senhas: ";
    cin >> quantidadeDigitos;

    vector<string> possibilidades = gerarSenhasForcaBruta(quantidadeDigitos);

    if (!possibilidades.empty()) {
        cout << "\nPossiveis senhas de " << quantidadeDigitos << " digitos:" << endl;
        for (const string& senha : possibilidades) {
            cout << senha << endl;
        }
        cout << "\nTotal de possibilidades geradas: " << possibilidades.size() << endl;
    }

    return 0;
}

File: Gerador de Senha/senha.py
def gerar_senhas_forca_bruta(num_digitos):
  """
  Gera todas as combinações possíveis de senhas com o número de dígitos especificado pelo usuário.

  Args:
    num_digitos: Um inteiro representando o número de dígitos da senha desejada.

  Returns:
    Uma lista de strings, onde cada string é uma possível senha.
  """
  if not isinstance(num_digitos, int) or num_digitos <= 0:
    return "Por favor, digite um número inteiro positivo para a quantidade de dígitos."

  senhas = []
  limite_superior = 10 ** num_digitos
  formato = f"0{num_digitos}d"  # Cria um formato para zfill com o número correto de dígitos

  for i in range(limite_superior):
    senha = format(i, formato)
    senhas.append(senha)
  return senhas

if __name__ == "__main__":
  try:
    quantidade_digitos = int(input("Digite a quantidade de dígitos desejada para as senhas: "))
    possibilidades = gerar_senhas_forca_bruta(quantidade_digitos)

    if isinstance(possibilidades, str):
      print(possibilidades)
    else:
      print(f"\nPossíveis senhas de {quantidade_digitos} dígitos:")
      for senha in possibilidades:
        print(senha)
      print(f"\nTotal de possibilidades geradas: {len(possibilidades)}")

  except ValueError:
    print("Entrada inválida. Por favor, digite um número inteiro.")